documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,452 @@
+This list mirrors Kafka permissions and should be the same for any deployment.
+
+There is an link:https://raw.githubusercontent.com/strimzi/strimzi-kafka-oauth/master/oauth-keycloak-authorizer/etc/authorization-scopes.json[authorization-scopes.json] file containing the authorization scopes that can be imported, so that they do not have to be manually entered for every new `Authorization Services` enabled client.
+In order to import `authorization-scopes.json` into a new client, first make sure the new client is `Authorization Enabled` and saved. Then, click on the `Authorization` tab and use the `Import` to import the file. Afterwards, if you select the `Authorization Scopes` you will see the loaded scopes.
+For this example the authorization scopes have already been imported as part of the realm import.
+
+Under the `Policies` sub-tab there are filters that match sets of users.
+Users can be explicitly listed, or they can be matched based on the Roles, or Groups they are assigned.
+Policies can even be programmatically defined using JavaScript where logic can take into account the context of the client session - for example, client ip (that is client ip of the Kafka client).
+
+Then, finally, there is the `Permissions` sub-tab, which defines 'role bindings' where `resources`, `authorization scopes` and `policies` are tied together to apply a set of permissions on specific resources for certain users.
+
+Each `permission` definition can have a nice descriptive name which can make it very clear what kind of access is granted to which users.
+For example:
+
+    Dev Team A can write to topics that start with x_ on any cluster
+
+    Dev Team B can read from topics that start with x_ on any cluster
+    Dev Team B can update consumer group offsets that start with x_ on any cluster
+
+    ClusterManager of my-cluster Group has full access to cluster config on my-cluster
+    ClusterManager of my-cluster Group has full access to consumer groups on my-cluster
+    ClusterManager of my-cluster Group has full access to topics on my-cluster
+
+If we take a closer look at the `Dev Team A can write ...` permission definition, we see that it combines a resource called `Topic:x_*`, scopes `Describe` and `Write`, and `Dev Team A` policy.
+If we click on the `Dev Team A` policy, we see that it matches all users that have a realm role called `Dev Team A`.
+
+Similarly, the `Dev Team B ...` permissions perform matching using the `Dev Team B` policy which also uses realm role to match allowed users - in this case those with realm role `Dev Team B`.
+The `Dev Team B ...` permissions grant users `Describe` and `Read` on `Topic:x_*`, and `Group:x_*` resources, effectively giving matching users and clients the ability to read from topics, and update the consumed offsets for topics and consumer groups that have names starting with 'x_'.
+
+## Targeting Permissions - Clients and Roles vs. Users and Groups
+
```

Comment:
Whether you use group or role policies to control access is dependent on the type of access. Generally, you will use roles for client access, and groups for user access.

Reply:
I don't think this is quite true. Users may easily decide to not use groups at all, and to only use roles for all types of accounts.